addons/website_version/models/website.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+import json
+from openerp import fields, models, api
+from openerp.http import request
+import random
+
+
+class NewWebsite(models.Model):
+
+    _inherit = "website"
+
+    tuto_google_analytics = fields.Boolean("How to get GA key and View ID")
+    google_analytics_view_id = fields.Char('View ID')
+    google_management_authorization = fields.Char('Google authorization')
+
+    @api.model
+    def get_current_version(self, context=None):
+        snap = request.env['website_version.version']
+        version_id = request.context.get('version_id')
+
+        if not version_id:
+            request.context['version_id'] = 0
+            return (0, '')
+        return (version_id, snap.browse(version_id).name)
+
+    @api.model
+    def get_current_website(self):
+        website = super(NewWebsite, self).get_current_website()
+        #We just set the cookie for the first visit
+        if 'website_version_experiment' in request.httprequest.cookies:
+            EXP = json.loads(request.httprequest.cookies.get('website_version_experiment'))
+        else:
+            EXP = request.context.get('website_version_experiment', {})
+            exps = self.env["website_version.experiment"].search([('state', '=', 'running'), ('website_id.id', '=', website.id)])
+            for exp in exps:
+                if not exp.google_id in EXP:
```

Comment:
You should include that condition in the search, as no else clause is used, no need to fetch data you won't use.